src/query.rs

```diff
@@ -59,21 +56,22 @@ pub fn q_once<'sqlite, 'schema, 'query>
     // TODO: validate inputs.
     let parsed = parse_find_string(query)?;
     let algebrized = algebrize(schema, parsed);
-    let projection = Projection::Star;
-    let select = cc_to_select(projection, algebrized.cc);
-    let SQLQuery { sql, args } = select.to_sql_query()?;
+    let select = query_to_select(algebrized);
+    let SQLQuery { sql, args } = select.query.to_sql_query()?;
 
-    /*
     let mut statement = sqlite.prepare(sql.as_str())?;
 
-    let mut rows = if args.is_empty() {
+    let rows = if args.is_empty() {
         statement.query(&[])?
     } else {
-        statement.query_named(args.map(|(k, v)| (k.as_str(), &v)))?
+        let refs: Vec<(&str, &ToSql)> =
+            args.iter()
+                .map(|&(ref k, ref v)| (k.as_str(), v as &ToSql))
+                .collect();
+        statement.query_named(refs.as_slice())?
     };
-    */
-
-
 
-    Ok(QueryResults::Scalar(Some(TypedValue::Boolean(true))))
+    select.projector
```

Comment:
Is this commit out of order?  Check before landing please, although this is often just a Github UI foible.